src/modules/products/templates/Product-display-wrapper/index.tsx

```diff
@@ -0,0 +1,44 @@
+'use client';
+
+import { PricedProduct } from '@medusajs/medusa/dist/types/pricing';
+import { Container } from '@medusajs/ui';
+import ImageGallery from '@modules/products/components/image-gallery';
+import Image from 'next/image';
+import { notFound } from 'next/navigation';
+import { useState } from 'react';
+
+type ProductDisplayWrapperProps = {
+  product: PricedProduct;
+};
+
+const ProductDisplayWrapper = ({ product }: ProductDisplayWrapperProps) => {
+  if (product.images?.length === 0) {
+    return notFound();
```

Comment:
You should not return notFound when there are no images for the product. Instead, when there are no images, always use fallback image found in `public/images/fallback.png`

Reply:
Example: `/src/modules/home/components/hero/index.tsx`